SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -580,16 +581,20 @@ void USpatialActorChannel::SetChannelActor(AActor* InActor)
 	EntityId = NetDriver->GetEntityRegistry()->GetEntityIdFromActor(InActor);
 
 	// If the entity registry has no entry for this actor, this means we need to create it.
-	if (EntityId == 0)
+	if (EntityId == SpatialConstants::INVALID_ENTITY_ID)
 	{
 		bCreatingNewEntity = true;
-		Sender->SendReserveEntityIdRequest(this);
+		TryAllocateEntityId();
 	}
 	else
 	{
 		UE_LOG(LogSpatialActorChannel, Log, TEXT("Opened channel for actor %s with existing entity ID %lld."), *InActor->GetName(), EntityId);
 
-		// Inform USpatialNetDriver of this new actor channel/entity pairing
+		if (NetDriver->PackageMap->IsEntityIdPendingCreation(EntityId))
```

Comment:
Assign the PackageMap to a variable since you use the NetDriver to get to it twice